common/logisticspipes/proxy/specialtankhandler/AETankHandler.java

```diff
@@ -0,0 +1,138 @@
+package logisticspipes.proxy.specialtankhandler;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import cpw.mods.fml.common.Loader;
+import appeng.api.AEApi;
+import appeng.api.implementations.tiles.ITileStorageMonitorable;
+import appeng.api.config.Actionable;
+import appeng.api.networking.IGridHost;
+import appeng.api.networking.IGridNode;
+import appeng.api.networking.security.IActionHost;
+import appeng.api.networking.security.MachineSource;
+import appeng.api.storage.IMEMonitor;
+import appeng.api.storage.IStorageMonitorable;
+import appeng.api.util.AECableType;
+import appeng.api.storage.data.IAEFluidStack;
+import logisticspipes.interfaces.ISpecialTankAccessHandler;
+import logisticspipes.utils.FluidIdentifier;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraftforge.common.util.ForgeDirection;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidTankInfo;
+import net.minecraftforge.fluids.IFluidHandler;
+import buildcraft.factory.TileTank;
+
+public class AETankHandler implements ISpecialTankAccessHandler {
+
+	@Override
+	public boolean init() {
+		return true;
+	}
+
+	@Override
+	public boolean isType(TileEntity tile) {
+		return tile instanceof ITileStorageMonitorable;
+	}
+
+	@Override
+	public List<TileEntity> getBaseTilesFor(TileEntity tile) {
```

Comment:
This should contain the controller and not the interface. It is used to identify If LP is connected to the same fluid providing structure to support a provider pipe and a supplier pipe on the same structure without causing a loop.